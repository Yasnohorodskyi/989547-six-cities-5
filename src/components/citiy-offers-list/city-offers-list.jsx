import React from "react";
import SortOffersComponent from "../sort-offers-component/sort-offers-component";
import propTypes from "prop-types";
import {connect} from "react-redux";
import MainCard from "../offer-card-city/offer-card-city";

const CityOffersList = (props) => {
  const {offers, activeCity} = props;

  return (
    <section className="cities__places places">
      <h2 className="visually-hidden">Places</h2>
      <b className="places__found">{offers.length} places to stay in {activeCity}</b>
      <SortOffersComponent />
      <div className="cities__places-list places__list tabs__content">
        {offers.map((offer) => <MainCard key={offer.id} offer={offer}/>)}
      </div>
    </section>
  );
};

const mapStateToProps = ({DATA}) => ({
  activeCity: DATA.activeCity
});

CityOffersList.propTypes = {
  offers: propTypes.array.isRequired,
  activeCity: propTypes.string.isRequired
};

export {CityOffersList};
export default connect(mapStateToProps)(CityOffersList);

