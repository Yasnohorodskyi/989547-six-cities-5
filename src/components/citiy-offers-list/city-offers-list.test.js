import React from "react";
import renderer from "react-test-renderer";

import {CityOffersList} from "./city-offers-list";

const activeCity = `Amsterdam`;
const offers = [
  {
    id: 123
  },
  {
    id: 124
  },
  {
    id: 125
  }
];

it(`Should City offers list render correctly`, () => {
  const tree = renderer
  .create(<CityOffersList
    activeCity={activeCity}
    offers={offers}
  />).toJSON();

  expect(tree).toMatchSnapshot();
});
