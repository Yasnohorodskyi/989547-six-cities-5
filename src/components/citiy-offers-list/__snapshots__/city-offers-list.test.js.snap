// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should City offers list render correctly 1`] = `
<section
  className="cities__places places"
>
  <h2
    className="visually-hidden"
  >
    Places
  </h2>
  <b
    className="places__found"
  >
    2
     places to stay in 
    Amsterdam
  </b>
  <form
    action="#"
    className="places__sorting"
    method="get"
  >
    <span
      className="places__sorting-caption"
    >
      Sort by 
    </span>
    <span
      className="places__sorting-type"
      onClick={[Function]}
      tabIndex="0"
    >
      <svg
        className="places__sorting-arrow"
        height="4"
        width="7"
      >
        <use
          xlinkHref="#icon-arrow-select"
        />
      </svg>
    </span>
    
  </form>
  <div
    className="cities__places-list places__list tabs__content"
  >
    <article
      className="cities__place-card place-card"
      onMouseOver={[Function]}
    >
      
      <div
        className="cities__image-wrapper place-card__image-wrapper"
      >
        <a
          href="/offer/123"
          onClick={[Function]}
        >
          <img
            className="place-card__image"
            height="200"
            src="https://picsum.photos/210"
            width="260"
          />
        </a>
      </div>
      <div
        className="place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              444
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            className="place-card__bookmark-button button place-card__bookmark-button--active"
            onClick={[Function]}
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height={19}
              width={18}
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "80%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <a
            href="/offer/123"
            onClick={[Function]}
          >
            Bla bla bla
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          Room
        </p>
      </div>
    </article>
    <article
      className="cities__place-card place-card"
      onMouseOver={[Function]}
    >
      
      <div
        className="cities__image-wrapper place-card__image-wrapper"
      >
        <a
          href="/offer/124"
          onClick={[Function]}
        >
          <img
            className="place-card__image"
            height="200"
            src="https://picsum.photos/210"
            width="260"
          />
        </a>
      </div>
      <div
        className="place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              445
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            className="place-card__bookmark-button button place-card__bookmark-button--active"
            onClick={[Function]}
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height={19}
              width={18}
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "80%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <a
            href="/offer/124"
            onClick={[Function]}
          >
            Hello, mello, shmello
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          House
        </p>
      </div>
    </article>
  </div>
</section>
`;
